lambdas/services/data_collection_service.py

```diff
@@ -0,0 +1,324 @@
+import hashlib
+import os
+from collections import Counter, defaultdict
+from datetime import datetime
+
+import polars as pl
+from enums.supported_document_types import SupportedDocumentTypes
+from models.statistics import (
+    ApplicationData,
+    OrganisationData,
+    RecordStoreData,
+    StatisticData,
+)
+from services.base.cloudwatch_logs_query_service import CloudwatchLogsQueryService
+from services.base.dynamo_service import DynamoDBService
+from services.base.s3_service import S3Service
+from utils.audit_logging_setup import LoggingService
+from utils.cloudwatch_logs_query import (
+    CloudwatchLogsQueryParams,
+    LloydGeorgeRecordsDeleted,
+    LloydGeorgeRecordsDownloaded,
+    LloydGeorgeRecordsStored,
+    LloydGeorgeRecordsViewed,
+    UniqueActiveUserIds,
+)
+from utils.common_query_filters import UploadCompleted
+from utils.utilities import flatten, get_file_key_from_s3_url
+
+logger = LoggingService(__name__)
+
+
+class DataCollectionService:
+    def __init__(self):
+        self.workspace = os.environ["WORKSPACE"]
+        self.output_table_name = os.environ["STATISTICS_TABLE"]
+
+        self.logs_query_service = CloudwatchLogsQueryService()
+        self.dynamodb_service = DynamoDBService()
+        self.s3_service = S3Service()
+
+        one_day = 60 * 60 * 24
+        time_now = int(datetime.now().timestamp())
+
+        self.collection_start_time = time_now - one_day
+        self.collection_end_time = time_now
+        self.today_date = datetime.today().strftime("%Y%m%d")
+
+    def collect_all_data_and_write_to_dynamodb(self):
+        time_period_human_readable = (
+            f"{datetime.fromtimestamp(self.collection_start_time)}"
+            f" ~ {datetime.fromtimestamp(self.collection_end_time)}"
+        )
+        logger.info(f"Collecting data between {time_period_human_readable}.")
+
+        all_statistic_data = self.collect_all_data()
+        logger.info("Finished collecting data. Will output to dynamodb table.")
+
+        self.write_to_local_dynamodb_table(all_statistic_data)
+        logger.info("Data collection completed.", {"Result": "Successful"})
+
+    def collect_all_data(self) -> list[StatisticData]:
+        dynamodb_scan_result = self.scan_dynamodb_tables()
+        s3_list_objects_result = self.get_all_s3_files_info()
+
+        record_store_data = self.get_record_store_data(
+            dynamodb_scan_result, s3_list_objects_result
+        )
+        organisation_data = self.get_organisation_data(dynamodb_scan_result)
+        application_data = self.get_application_data()
+
+        return record_store_data + organisation_data + application_data
+
+    def write_to_local_dynamodb_table(self, all_statistic_data: list[StatisticData]):
```

Comment:
Why have we included 'local' here?

Reply:
sorry, this was a legacy name when I built this with a locally hosted dynamodb
renamed to `write_to_dynamodb_table` now